test/suites/scheduling/suite_test.go

```diff
@@ -0,0 +1,514 @@
+						LabelSelector: &metav1.LabelSelector{
+							MatchLabels: labels,
+						},
+						TopologyKey: corev1.LabelHostname,
+					}},
+					InitContainers:       initContainers,
+					ResourceRequirements: containerRequirements,
+				})
+				env.ExpectCreated(nodePool, nodeClass, dsBufferPod, pod)
+				env.EventuallyExpectHealthy(pod)
+				node := env.ExpectCreatedNodeCount("==", 1)[0]
+				Expect(node.ObjectMeta.GetLabels()[v1alpha2.LabelSKUCPU]).To(Equal(expectedNodeCPU))
+			},
+			Entry("sidecar requirements + later init requirements do exceed container requirements", "8", corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("400m")},
+			}, ephemeralInitContainer(corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("300m")},
+			}), corev1.Container{
+				RestartPolicy: lo.ToPtr(corev1.ContainerRestartPolicyAlways),
+				Resources: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("350m")},
+				},
+			}, ephemeralInitContainer(corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("1")},
+			})),
+			Entry("sidecar requirements + later init requirements do not exceed container requirements", "4", corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("400m")},
+			}, ephemeralInitContainer(corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("300m")},
+			}), corev1.Container{
+				RestartPolicy: lo.ToPtr(corev1.ContainerRestartPolicyAlways),
+				Resources: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("350m")},
+				},
+			}, ephemeralInitContainer(corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("300m")},
+			})),
+			Entry("init container requirements exceed all later requests", "8", corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("400m")},
+			}, corev1.Container{
+				RestartPolicy: lo.ToPtr(corev1.ContainerRestartPolicyAlways),
+				Resources: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("100m")},
+				},
+			}, ephemeralInitContainer(corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("1500m")},
+			}), corev1.Container{
+				RestartPolicy: lo.ToPtr(corev1.ContainerRestartPolicyAlways),
+				Resources: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("100m")},
+				},
+			}),
+		)
+		It("should provision a right-sized node when a pod has InitContainers (mixed resources)", func() {
+			if env.K8sMinorVersion() < 29 {
+				Skip("native sidecar containers are only enabled on EKS 1.29+")
```

Comment:
```suggestion
				Skip("native sidecar containers are only enabled on AKS 1.29+")
```

Is this supported on AKS? I guess the test would fail if it wasn't